ostree/ostree_dest.go

```diff
@@ -389,7 +390,7 @@ func (d *ostreeImageDestination) PutSignatures(signatures [][]byte) error {
 	return nil
 }
 
-func (d *ostreeImageDestination) Commit() error {
+func (d *ostreeImageDestination) Commit(ctx context.Context) error {
```

Comment:
`importBlob`, `fixFiles`, `generateTarSplitMetadata`, and maybe more, can handle many-gigabyte files or directory structures and should be cancelable.